sklearn/linear_model/_coordinate_descent.py

```diff
@@ -435,6 +435,18 @@ def enet_path(X, y, *, l1_ratio=0.5, eps=1e-3, n_alphas=100, alphas=None,
     :ref:`examples/linear_model/plot_lasso_coordinate_descent_path.py
     <sphx_glr_auto_examples_linear_model_plot_lasso_coordinate_descent_path.py>`.
     """
+    X_offset_param = params.pop('X_offset', None)
+    X_scale_param = params.pop('X_scale', None)
+    tol = params.pop('tol', 1e-4)
+    max_iter = params.pop('max_iter', 1000)
+    random_state = params.pop('random_state', None)
+    selection = params.pop('selection', 'cyclic')
+
+    params.pop('fit_intercept', None)
+    params.pop('normalize', None)
```

Comment:
there seems to be many places in the tests where `fit_intercept` and and `normalize` are being passed into enet_path - changing `**params` to keyword args will necessitate a refactor of all tests that pass in fit_intercept and normalize

should we still proceed in this direction?

Reply:
Indeed, `fit_intercept` and `normalize` are not used in the path. They used to prepare the data before `fit`.
I assume that we should raise the error if one tries to pass these parameters since the `path` is not doing anything regarding these parameters.

However, we should add in the documentation that `lasso_path` and `lasso_path` expect centred and normalized data. If this is not the case, I don't know if we should stipulate to use the `Lasso` and `ElasticNet` estimators instead. @agramfort can give some highlight here.